ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/kre/helpers/KreCaseChangeHelperTest.java

```diff
@@ -0,0 +1,67 @@
+package com.kayako.sdk.android.k5.kre.helpers;
+
+import com.kayako.sdk.android.k5.kre.base.KreSubscription;
+import com.kayako.sdk.android.k5.kre.data.Change;
+import static junit.framework.Assert.assertNull;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+public class KreCaseChangeHelperTest {
+
+    private static final String EVENT_CHANGE = "CHANGE";
+    private static final String MESSAGE = "message";
+
+    @Mock
+    private KreSubscription kreSubscription;
+
+    @Mock
+    private RawCaseChangeListener rawCaseChangeListener;
+
+    @Captor
+    private ArgumentCaptor<Change> changeArgumentCaptor;
+
+    @Test
+    public void addRawCaseChangeListener() {
+        //Arrange
+        final String event = "testEvent";
+        final String jsonBody = "{\n" +
```

Comment:
❗️ This snippet of code represents a JSON build via `String` concatenations. Consider relying on a dedicated library for JSON, such as Java API for JSON (JSR 374), GSON, Jackson, etc. This way the code eliminates the risk of using corrupted JSON data, mitigate typos and increase readability.  

**[More details](https://confluence.devfactory.com/x/DwEGF)**
          
## The solution in this case

```JAVA
// Add Java JSON API via Gradle
https://mvnrepository.com/artifact/javax.json/javax.json-api
gradle.properties   
    java_json_version = 1.1.2
 
build.gradle
    compile group: 'javax.json', name: 'javax.json-api', version: java_json_version



import javax.json.Json;
import javax.json.JsonObject;
...
final String jsonBody = Json.createObjectBuilder()
        .add("resource_url", "https://kayako-mobile-testing.kayako.com/api/v1/cases/posts/2191")
        .add("resource_type", "post")
        .add("resource_id", 2191)
         ...
        .build()
        .toString();

```
 

<a id="6fb17600-54f2-11e8-9db9-b7d507f482ce" href="none" codereview-uuid-reserved-tag />

Reply:
🚫  The snippet of code relies on hard-coded newline separator. Depending on the operating system (e.g., Linux, OS X, Windows) the line endings may differ and this may lead to inconsistent results. For example, if we get this wrong, then we will mess up the ability to exchange data with other applications. So, it is recommended to rely on Java support, which is capable to provide the platform-independent line separator.  

> **Warning**: Implementing this suggestion may have deeper implications or may require bigger refactoring, therefore it is important to make sure it is agreed at the project level.

**[More details](https://confluence.devfactory.com/x/1AgGF)**
          
## The solution in this case

```Java  

private static final String NEW_LINE = System.lineSeparator();
...
 final String jsonBody = "{" + NEW_LINE 

```
**Note**: In case the suggestion will not be implemented consider extracting the newline separator in a constant. 

<a id="36547190-54f3-11e8-9db9-b7d507f482ce" href="none" codereview-uuid-reserved-tag />